### PR TITLE
Add helpful "getting started" documentation

### PR DESCRIPTION
**Pull request by DoubleCouponDay**
I recently struggled using entityframework6.npgsql and the documentation was not helpful. This updated documentation will help future users from running into the same problems I did. #100

### CODE REVIEW

File: doc/index.md [hunk 1]
```diff
@@ -6,6 +6,21 @@ title: Entity Framework 6
 Npgsql has an Entity Framework 6 provider. You can use it by installing the
 [EntityFramework6.Npgsql](https://www.nuget.org/packages/EntityFramework6.Npgsql/) nuget.
 
+## Basic Usage ##
+Create an inheritor of DbConfiguration in the same assembly as your entity framework DbContext. Ensure that you configure provider services, a provider factory, a default connection factory as shown below:
+
+```
```
Comment (austindrenski) [inline doc/index.md]: This should have a trailing `c#` for syntax highlighting.

File: doc/index.md [hunk 2]
```diff
@@ -6,6 +6,21 @@ title: Entity Framework 6
 Npgsql has an Entity Framework 6 provider. You can use it by installing the
 [EntityFramework6.Npgsql](https://www.nuget.org/packages/EntityFramework6.Npgsql/) nuget.
 
+## Basic Usage ##
```
Comment (austindrenski) [inline doc/index.md]: This section seems to be more about configuration than usage. Propose changing usage to configuration.

File: doc/index.md [hunk 3]
```diff
@@ -6,6 +6,21 @@ title: Entity Framework 6
...
+Create an inheritor of DbConfiguration in the same assembly as your entity framework DbContext. Ensure that you configure provider services, a provider factory, a default connection factory as shown below:
```
Comment (austindrenski) [inline doc/index.md]: "Create an inheritor of DbConfiguration" should be rephrased. Something like "Define a class that inherits from `DbConfiguration`.

Note the inline code formatting for `DbConfiguration` and `DbContext`.

File: doc/index.md [hunk 4]
```diff
@@ -6,6 +6,21 @@ title: Entity Framework 6
...
+
+```
+class NpgSqlConfiguration : DbConfiguration
+{
+    public NpgSqlConfiguration()
+    {
+        SetProviderFactory("Npgsql", NpgsqlFactory.Instance);
```
Comment (austindrenski) [inline doc/index.md]: For consistency, how about using the (optional) named parameters for all of these?

File: doc/index.md (hunk 1 above)
Comment (YohDeadfall) [reply doc/index.md]: `csharp` is okay too.

File: doc/index.md (hunk 2 above)
Comment (DoubleCouponDay) [reply doc/index.md]: resolved.

File: doc/index.md (hunk 3 above)
Comment (DoubleCouponDay) [reply doc/index.md]: resolved.

File: doc/index.md (hunk 1 above)
Comment (DoubleCouponDay) [reply doc/index.md]: resolved.

File: doc/index.md (hunk 4 above)
Comment (DoubleCouponDay) [reply doc/index.md]: resolved.

**Comment by DoubleCouponDay** [comment]
changes requested have been implemented.

**Comment by roji** [comment]
Looks good to me, @austindrenski feel free to merge.

**Comment by austindrenski** [comment]
@DoubleCouponDay Thank you for your contribution!

**Comment by roji** [comment]
Updated docs, this change is online.

**Pull request merged**